FreeRTOS/Demo/CORTEX_MPU_M3_NUCLEO_L152RE_GCC/Demo/reg_tests.c

```diff
@@ -0,0 +1,332 @@
+                            { 0, 0, 0 },
+                            { 0, 0, 0 }
+                        }
+};
+
+    xTaskCreateRestricted( &( xRegTest1TaskParameters ), NULL );
+    xTaskCreateRestricted( &( xRegTest2TaskParameters ), NULL );
+    xTaskCreateRestricted( &( xRegTest3TaskParameters ), NULL );
+    xTaskCreateRestricted( &( xRegTest4TaskParameters ), NULL );
+    xTaskCreateRestricted( &( xCheckTaskParameters ), NULL );
+}
+/*-----------------------------------------------------------*/
+
+static void prvRegTest1Task( void * pvParameters )
+{
+    /* Although the reg tests are written in assembly, its entry
+     * point is written in C for convenience of checking that the
+     * task parameter is being passed in correctly. */
+    if( pvParameters == REG_TEST_TASK_1_PARAMETER )
+    {
+        /* Start the part of the test that is written in assembler. */
+        vRegTest1Asm();
+    }
+
+    /* The following line will only execute if the task parameter
+     * is found to be incorrect. The check task will detect that
+     * the reg test loop counter is not being incremented and flag
+     * an error. */
+    vTaskDelete( NULL );
+}
+/*-----------------------------------------------------------*/
+
+static void prvRegTest2Task( void * pvParameters )
+{
+    /* Although the reg tests are written in assembly, its entry
+     * point is written in C for convenience of checking that the
+     * task parameter is being passed in correctly. */
+    if( pvParameters == REG_TEST_TASK_2_PARAMETER )
+    {
+        /* Start the part of the test that is written in assembler. */
+        vRegTest2Asm();
+    }
+
+    /* The following line will only execute if the task parameter
+     * is found to be incorrect. The check task will detect that
+     * the reg test loop counter is not being incremented and flag
+     * an error. */
+    vTaskDelete( NULL );
+}
+/*-----------------------------------------------------------*/
+
+static void prvRegTest3Task( void * pvParameters )
+{
+    /* Although the reg tests are written in assembly, its entry
+     * point is written in C for convenience of checking that the
+     * task parameter is being passed in correctly. */
+    if( pvParameters == REG_TEST_TASK_3_PARAMETER )
+    {
+        /* Start the part of the test that is written in assembler. */
+        vRegTest3Asm();
+    }
+
+    /* The following line will only execute if the task parameter
+     * is found to be incorrect. The check task will detect that
+     * the reg test loop counter is not being incremented and flag
+     * an error. */
+    vTaskDelete( NULL );
+}
+/*-----------------------------------------------------------*/
+
+static void prvRegTest4Task( void * pvParameters )
+{
+    /* Although the reg tests are written in assembly, its entry
+     * point is written in C for convenience of checking that the
+     * task parameter is being passed in correctly. */
+    if( pvParameters == REG_TEST_TASK_4_PARAMETER )
```

Comment:
I like the idea of comparing the void pointer with a macro. That's probably one of the better ways to compare if the type cannot be passed to the function.